From fd921434711d4f60767aed583d15b5ef33651830 Mon Sep 17 00:00:00 2001
From: Rob Hoes <rob.hoes@citrix.com>
Date: Thu, 22 Aug 2013 14:24:31 +0100
Subject: [PATCH 33/34] libxl: an 'unknown' shutdown reason may manifest as
 255, not -1

This should probably be fixed in libxl, consider this a workaround.

Although xen uses -1 as an 'unknown' shutdown reason, when this
is marshalled into a getdomaininfo domctl, it is masked with 0xff
and treated as an unsigned value.

Signed-off-by: David Scott <dave.scott@eu.citrix.com>
Signed-off-by: Rob Hoes <rob.hoes@citrix.com>
---
 tools/libxl/libxl_types.idl |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/libxl/libxl_types.idl b/tools/libxl/libxl_types.idl
index 58fd103..4f87781 100644
--- a/tools/libxl/libxl_types.idl
+++ b/tools/libxl/libxl_types.idl
@@ -125,7 +125,7 @@ libxl_scheduler = Enumeration("scheduler", [
 
 # Consistent with SHUTDOWN_* in sched.h (apart from UNKNOWN)
 libxl_shutdown_reason = Enumeration("shutdown_reason", [
-    (-1, "unknown"),
+    (255, "unknown"),
     (0, "poweroff"),
     (1, "reboot"),
     (2, "suspend"),
-- 
1.7.10.4

